src/olympia/devhub/tasks.py

```diff
@@ -265,6 +268,22 @@ def handle_file_validation_result(results, file_id, *args):
     return FileValidation.from_json(file_, results)
 
 
+def insert_validation_message(results, type_='error', message='', msg_id='',
+                              compatibility_type=None):
+    messages = results['messages']
+    messages.insert(0, {
+        'tier': 1,
+        'type': type_,
+        'id': ['validation', 'messages', msg_id],
+        'message': message,
+        'description': [],
+        'compatibility_type': compatibility_type,
+    })
+    # Need to increment 'errors' or 'warnings' count, so add an extra 's' after
+    # the type_ to increment the right entry.
+    results['{}s'.format(type_)] += 1
```

Comment:
I probably would have used a mapping for the `error` -> `errors` stuff but that works too 😆